contracts/contracts/coordination/TACoChildApplication.sol

```diff
@@ -123,6 +130,56 @@ contract TACoChildApplication is ITACoRootToChild, ITACoChildApplication, Initia
         emit OperatorConfirmed(stakingProvider, _operator);
         rootApplication.confirmOperatorAddress(_operator);
     }
+
+    /**
+     * @notice Return the length of the array of staking providers
+     */
+    function getStakingProvidersLength() external view returns (uint256) {
+        return stakingProviders.length;
+    }
+
+    /**
+     * @notice Get the value of authorized tokens for active providers as well as providers and their authorized tokens
+     * @param _startIndex Start index for looking in providers array
+     * @param _maxStakingProviders Max providers for looking, if set 0 then all will be used
+     * @return allAuthorizedTokens Sum of authorized tokens for active providers
+     * @return activeStakingProviders Array of providers and their authorized tokens.
+     * Providers addresses stored as uint256
+     * @dev Note that activeStakingProviders[0] is an array of uint256, but you want addresses.
+     * Careful when used directly!
+     */
+    function getActiveStakingProviders(
```

Comment:
In `nucypher-porter` and `nucypher` we currently do sampling using the same function on `TACoApplication` contract instead of `TACoChildApplication`. With this change, instead, we can sample using `TACoChildApplication`? Is there any difference between sampling using root vs child, other than using different chains? Is one potentially more up-to-date than the other?

I'm asking because, at the moment, Porter doesn't use/need any polygon endpoint, it only needs an eth endpoint. If using `TACoChildApplication` is better for some reason, then Porter will need to be updated to use polygon.

Reply:
yep, that's alternative to method in TACo root app until bot to sync from polygon to mainnet will be ready. 
Thing is - sampling returns only confirmed providers/operators. And without bot - all providers in root app will be unconfirmed. And considering that child has all data already - that's easiest option to start with